projects/plugins/jetpack/extensions/shared/external-media/sources/google-photos/index.js

```diff
@@ -1,29 +1,56 @@
+import moment from 'moment';
 import { useEffect, useState } from 'react';
 import MediaLoadingPlaceholder from '../../media-browser/placeholder';
+import { getGooglePhotosPickerCachedSessionId } from '../../media-service';
 import { MediaSource } from '../../media-service/types';
 import withMedia from '../with-media';
 import GooglePhotosAuth from './google-photos-auth';
 import GooglePhotosMedia from './google-photos-media';
 import GooglePhotosPickerButton from './google-photos-picker-button';
 
 function GooglePhotos( props ) {
-	const { isAuthenticated, pickerSession, createPickerSession, getPickerStatus } = props;
+	const {
+		isAuthenticated,
+		pickerSession,
+		createPickerSession,
+		fetchPickerSession,
+		getPickerStatus,
+	} = props;
+	const [ cachedSessionId ] = useState( getGooglePhotosPickerCachedSessionId() );
+	const [ isCashedSessionChecked, setIsCashedSessionChecked ] = useState( false );
 	const [ pickerFeatureEnabled, setPickerFeatureEnabled ] = useState( null );
 	const isPickerSessionAccurate = pickerSession !== null && ! ( 'code' in pickerSession );
+	const isSessionExpired =
+		pickerSession?.expireTime && moment( pickerSession.expireTime ).isBefore( new Date() );
 
 	useEffect( () => {
 		getPickerStatus().then( feature => {
 			feature && setPickerFeatureEnabled( feature.enabled );
 		} );
-	}, [ getPickerStatus ] );
+		cachedSessionId &&
+			fetchPickerSession( cachedSessionId ).then( () => setIsCashedSessionChecked( true ) );
+	}, [ getPickerStatus, fetchPickerSession, cachedSessionId ] );
 
 	useEffect( () => {
-		if ( pickerFeatureEnabled && isAuthenticated && ! isPickerSessionAccurate ) {
+		if (
+			pickerFeatureEnabled &&
+			isCashedSessionChecked &&
```

Comment:
If the cache session id is null, it won't run createPickerSession.

Reply:
Thanks for catching it! It's fixed now!